### PR TITLE
pd: speed up reconnecting to a pd leader

### PR DESCRIPTION
**Pull request by overvenus**
Client may spend a lot to time to reconnect when the pd leader is isolated. Speed it up by adding a timeout.
Also it puts the previous leader at the last when reconnecting.

### CODE REVIEW

File: src/pd/util.rs [hunk 1]
```diff
@@ -389,16 +388,17 @@ pub fn try_connect_leader(
...
-    rand::thread_rng().shuffle(&mut indexes);
-
     let cluster_id = previous.get_header().get_cluster_id();
     let mut resp = None;
-    'outer: for i in indexes {
-        for ep in members[i].get_client_urls() {
+    // Try to connect to other members, then the previous leader.
+    'outer: for m in members
+        .into_iter()
+        .filter(|m| *m == previous_leader)
```
Comment (disksing) [inline src/pd/util.rs]: `==` should be `!=`?

Comment (siddontang) [reply src/pd/util.rs]: the filter will skip the previous leader, maybe fine here.

**Comment by siddontang** [comment]
LGTM

**Comment by disksing** [comment]
LGTM.

**Pull request merged**